src/scraper_net.cpp

```diff
@@ -55,7 +55,12 @@ bool CSplitBlob::RecvPart(CNode* pfrom, CDataStream& vRecv)
             return false;
         }
     } else {
-        if(pfrom)  pfrom->Misbehaving(SCRAPER_MISBEHAVING_NODE_BANSCORE / 5);
+        if(pfrom)
+        {
+            pfrom->Misbehaving(SCRAPER_MISBEHAVING_NODE_BANSCORE / 5);
+            LogPrintf("WARNING: CSplitBlob::RecvPart: Spurious part received from %s. Adding %u banscore.",
+                     pfrom->addr.ToString(), SCRAPER_MISBEHAVING_NODE_BANSCORE / 5);
+        }
         return error("Spurious part received!");
```

Comment:
If you extended the message, please remove this error message (redundant?).

Reply:
You are right.

Reply:
One question on that after looking at it carefully. The error message returned is not conditioned on the current existence of the CNode pfrom, unlike the banscore part (and my new logging message). They are not exactly duplicative.